src/ingest-pipeline/airflow/dags/undo_upload_reorganization.py

```diff
@@ -0,0 +1,276 @@
+    #   - This would mean that it did not create all of its children datasets.
+    #   - We don't have to worry about moving files around, since that's done in the second step.
+    #   - We just need to delete the children datasets, which is something that Pitt will do
+    # - Upload failed during second step
+    #   - Upload failed during data movement
+    #       - We definitely need to move data back. In this case, there should be a file somewhere that has the full list of dataset UUIDs
+    #       - We iterate over the dataset UUIDs, move data back to the upload under the appropriate data directory (what is in the original metadata)
+    #       - Expect that we receive a run id (e5ce5afb6653e5d88d1ba2fe1cb89aa9_reorganize.upload_2025-09-11T10:47:58.855034-04:00) as part of the config.
+    #       - Using the run id, we can read in the frozen file, and use the "new_uuid" field to move data back
+    #   - Upload failed during Upload status update
+    #       - Same steps as above
+    #   - Upload failed during dataset updates
+    #       - If we want to move data back we can, but it would probably be easiest to just bulk update/re-ingest metadata
+    #   - Upload failed during re-indexing
+    #       - Not really anything to fix here, just need to reindex things
+
+    def instantiate_factories():
+        # Need an EntityFactory
+        auth_tok = get_auth_tok(
+            **{
+                "crypt_auth_tok": encrypt_tok(
+                    airflow_conf.as_dict()["connections"]["APP_CLIENT_SECRET"]
+                ).decode()
+            }
+        )
+        return EntityFactory(
+            auth_tok,
+            instance=find_matching_endpoint(HttpHook.get_connection("entity_api_connection").host),
+        )
+
+    @task()
+    def check_conf(dag_run: DagRun):
+        if "prev_run_id" not in dag_run.conf or "upload_uuid" not in dag_run.conf:
+            print("Missing prev_run_id or upload_uuid in configuration.")
+            raise AirflowException("Missing required configuration: prev_run_id or upload_uuid")
+
+        ef = instantiate_factories()
+        upload = ef.get(dag_run.conf["upload_uuid"])
+        if upload.prop_dct["entity_type"] != "Upload":
+            print(
+                f"UUID {dag_run.conf['upload_uuid']} is not an Upload. Entity type is {upload.prop_dct['entity_type']}."
+            )
+            raise AirflowException(f"UUID {dag_run.conf['upload_uuid']} is not an Upload")
+
+        if not (_get_scratch_base_path() / dag_run.conf["prev_run_id"]).exists():
+            print(f"Could not find path {_get_scratch_base_path() / dag_run.conf['prev_run_id']}")
+            raise AirflowException(
+                f"Could not find path {_get_scratch_base_path() / dag_run.conf['prev_run_id']}"
+            )
+
+        dryrun = dag_run.conf.get("dryrun", False)
+        if dryrun:
+            print("DRYRUN MODE: Will only print paths, no actual data movement")
```

Comment:
I don't see much need of this message in the task, it doesn't hurt either.

Reply:
Removed